openedx/openedx-platform
# feat: add flags to studio for editors work

## Description

In order to support ongoing editor work we need to open blocks in the course_authoring MFE from studio. We are gating that rollout behind a flag. This work does that for each of the new blocks, as well as provides urls to do so.

## Supporting information

List of flags: 
new_core_editors.use_new_text_editor
new_core_editors.use_new_video_editor
new_core_editors.use_new_problem_editor

Documentation of those [flags](https://openedx.atlassian.net/wiki/spaces/PT/pages/3275325521/BD-14+Editor+Experience+Waffle+Flags)

Planned Removal Work: https://openedx.atlassian.net/browse/TNL-9370

## Testing instructions

Check out branch.
Restart LMS, remake static assets using `$ NO_PREREQ_INSTALL=1 paver update_assets lms` in  lms-shell
create 4 blocks in a unit in studio:
-Text
-Video
-Problem
-A block which is not one of these three block types (ORA, )

Test with all flags off (default): should see all editors function as normal.

Turn each flag on and see when that flag is turned on, clicking edit on that particular block redirects you to 
`<env_url for cousre_authoring>/ course/<course_id>/editor/<editor_type/<block_id>`

To turn a flag on, begin by navigating to http://localhost:18000/admin/waffle/flag/ and adding the `toggle_name` of the flag to the `flag name` field and turning visible to everyone to "yes."

ORA block still functions as normal.

## Other information

NOT WORTH MERGING TILL COURSE AUTHORING MFE HOLDS PAGES AT THOSE URLS
